src/Controller/CRUDController.php

```diff
@@ -1554,7 +1554,22 @@ private function checkParentChildAssociation(Request $request, object $object):
         $propertyAccessor = PropertyAccess::createPropertyAccessor();
         $propertyPath = new PropertyPath($this->admin->getParentAssociationMapping());
 
-        if ($parentAdmin->getObject($parentId) !== $propertyAccessor->getValue($object, $propertyPath)) {
+        $parentObject = $parentAdmin->getObject($parentId);
+        $objectParent = $propertyAccessor->getValue($object, $propertyPath);
+
+        $parentObjectMatches = $parentObject === $objectParent;
+
+        // $objectParent may be an array or a Collection when the parent association is many to many.
+        if (!$parentObjectMatches && is_iterable($objectParent)) {
+            foreach ($objectParent as $objectParentItem) {
+                $parentObjectMatches = $parentObject === $objectParentItem;
+                if ($parentObjectMatches) {
+                    break;
+                }
+            }
+        }
```

Comment:
Can you please extract this to a private method? I think it would be more understandable

Reply:
I can do that. This will make CRUDController contain even more private methods. Perhaps there is some static helper class that I could use?

Reply:
Hmm I don't remember one, I think it's fine here.

Reply:
I think this is quite hard, because the block you're quoting needs a lot of input variables: $parentAdmin, $parentId, $propertyAccessor, $object, $propertyPath. Either I make a new private method with all these variables as parameters, which is ugly, or I move some more code from `checkParentChildAssociation` to the new method. But this leaves `checkParentChildAssociation` with almost no code and the new method with all the code, so the split makes no difference in readability.

I moved a little less code and this is hopefully as readable as it can get.